### PR TITLE
Feature/create skill system

### PR DESCRIPTION
**Pull request by Corvus-inc**

### CODE REVIEW

**Comment by Corvus-inc** [comment]
Timer in skills mechanics  no react on pause. Need the issue for this the bug.

File: Assets/Scripts/Skills/SkillSystem.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                _healthSystem.Heal(valueHeal);
+            },
+            () =>
+            {
+                _continuesHealSkill = false;
+                Debug.Log("Heal Recharged!");
+            },
+            ref _continuesHealSkill
+            );
+        IsHeal.Invoke(_mSecForHealSkill);
```
Comment (pushkinman) [inline Assets/Scripts/Skills/SkillSystem.cs:49]:
```suggestion
        IsHeal?.Invoke(_mSecForHealSkill);
```

File: Assets/Scripts/Skills/SkillSystem.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            },
+            () =>
+            {
+                _player.IsBoostedSpeed = false;
+                _continuesBoostSkill = false;
+                Debug.Log("Speed Recharged!");
+            },
+            ref _continuesBoostSkill
+            );
+        IsBoostSpeed.Invoke(_mSecForBoostSkill);
```
Comment (pushkinman) [inline Assets/Scripts/Skills/SkillSystem.cs:67]:
```suggestion
        IsBoostSpeed?.Invoke(_mSecForBoostSkill);
```

File: Assets/Scripts/Skills/SkillSystem.cs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                _currentWeapon.StartDoubleDamage(20);
+            }, 
+            () =>
+            {
+                _continuesDamageSkill = false;
+                Debug.Log("Damage Recharged!");
+            },
+            ref _continuesDamageSkill
+            );
+        IsIncreaseDamage.Invoke(_mSecForDamageSkill);
```
Comment (pushkinman) [inline Assets/Scripts/Skills/SkillSystem.cs:85]:
```suggestion
        IsIncreaseDamage?.Invoke(_mSecForDamageSkill);
```

File: Assets/Scripts/Skills/SkillSystem.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+    private bool _continuesBoostSkill;
+    private bool _continuesDamageSkill;
+
+    public SkillSystem(HealthSystem healthSystem, PlayerCharacter player, Func<Weapon>  getWeapon)
+    {
+        _healthSystem = healthSystem;
+        _player = player;
+        _getWeapon = getWeapon;
+        _currentWeapon = getWeapon.Invoke();
+    }
```
Comment (pushkinman) [inline Assets/Scripts/Skills/SkillSystem.cs:32]: Fix errors in other places to match this new constructor
```suggestion
    public SkillSystem(HealthSystem healthSystem, PlayerCharacter player, WeaponSystem weaponSystem)
    {
        _healthSystem = healthSystem;
        _player = player;
        _weaponSystem = weaponSystem;
        _currentWeapon = _weaponSystem.GetCurrentWeapon();
    }
```

File: Assets/Scripts/Player/PlayerCharacter.cs [hunk 5]
```diff
@@ -9,39 +10,37 @@ public class PlayerCharacter : MonoBehaviour
...
     
     private void Update()
     {
         if (Input.GetKeyDown(KeyCode.Q))
         {
             //Move to Weapon system
-            _playerWeapon.ReturnAllBulletToSpawn();
-            _playerWeapon = WeaponSystem.SwitchWeapon();
+            PlayerWeapon.ReturnAllBulletToSpawn();
+            PlayerWeapon = WeaponSystem.SwitchWeapon();
```
Comment (pushkinman) [inline Assets/Scripts/Player/PlayerCharacter.cs:38]: Pay close attention to your role separation, the player does not need to know that the bullets need to be returned to spawn, that is the responsibility of WeaponSystem, so it show look like this
```suggestion
            PlayerWeapon = WeaponSystem.SwitchWeapon();
```

**Review by pushkinman — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Bullets/BaseBullet.cs [hunk 6]
```diff
@@ -15,19 +15,24 @@ public abstract class BaseBullet : MonoBehaviour, IBullet
...
+    private void Start()
+    {
+        gameObject.SetActive(false);
+    }
+
     public virtual void ActivatingBullet()
     {
         gameObject.SetActive(true);
-        IsActiveBullet?.Invoke();
+        OnActiveBullet();
```
[the remark this answers is not in the document]
Comment (Corvus-inc) [reply Assets/Scripts/Bullets/BaseBullet.cs:35]: these are the implementation features

**Pull request merged**